js_modules/dagit/packages/core/src/partitions/PartitionStatus.tsx

```diff
@@ -128,22 +130,27 @@ export const PartitionStatus: React.FC<{
     };
   }, [onSelect, selected, currentSelectionRange, getRangeSelection, toPartitionName]);
 
-  const selectedSpans = selected
-    ? assembleIntoSpans(partitionNames, (key) => selected.includes(key)).filter((s) => s.status)
-    : [];
+  const selectedSpans = React.useMemo(
+    () =>
+      selectedSet.size === 0
+        ? []
+        : selectedSet.size === partitionNames.length
+        ? [{startIdx: 0, endIdx: partitionNames.length - 1, status: true}]
+        : assembleIntoSpans(partitionNames, (key) => selectedSet.has(key)).filter((s) => s.status),
+    [selectedSet, partitionNames],
```

Comment:
Same question as above re: `partitionNames`

Reply:
Ahh this is a good q - these are memoized further up in the state, so I think it's best not to re-memoify them here.  I was curious about the Set case above though and did some testing - 

it turns out it's actually faster for a large array to build a new Set every time than attempt to JSON.stringify to compare:

```
array = (new Array(1000000)).fill('1234') 
d = performance.now(); JSON.stringify(array); console.log(performance.now() - d);
61.32737499475479
d = performance.now(); new Set(array); console.log(performance.now() - d);
14.73987501859665
```